### ISSUE TITLE
BigDecimal swagger rendering

### ISSUE THREAD

**Issue by jonnii** [micronaut-projects/micronaut-openapi#555]
### Expected Behavior

When applying a `@Schema(type = "string")` annotation to a field of BigDecimal it should change the `type` in the swagger to `string`.

### Actual Behaviour

The field instead lifts the BigDecimal type up as a schema type. 

### Steps To Reproduce

Below is an example of a diff where the annotation was added and the corresponding swagger diff that occurred. 

```
diff --git a/example/docs/openapi/swagger.yml b/example/docs/openapi/swagger.yml
index 5431a02..eae1922 100644
--- a/example/docs/openapi/swagger.yml
+++ b/example/docs/openapi/swagger.yml
@@ -356,6 +356,8 @@ components:
       properties:
         reason:
           type: string
+    BigDecimal:
+      type: string
     CreateUlidRequest:
       required:
       - ulid
@@ -466,8 +468,9 @@ components:
         enum:
           $ref: '#/components/schemas/Speed'
         numberAsString:
-          type: number
-          nullable: true
+          allOf:
+          - $ref: '#/components/schemas/BigDecimal'
+          - nullable: true
     Speed:
       type: string
       enum:
diff --git a/example/src/main/kotlin/com/nydig/micronaut/example/SerializationController.kt b/example/src/main/kotlin/com/nydig/micronaut/example/SerializationController.kt
index 4dbb659..6b3d107 100644
--- a/example/src/main/kotlin/com/nydig/micronaut/example/SerializationController.kt
+++ b/example/src/main/kotlin/com/nydig/micronaut/example/SerializationController.kt
@@ -56,6 +56,7 @@ class SerializationController {
         val missingThing: String?,
         val enum: Speed,
 
+        @field:Schema(type = "string")
         val numberAsString: BigDecimal?
     )
 
```

### Version

3.0.0

**Comment by jonnii** [comment]
For those looking for a work around it is possible to do the following:

```
@Schema(type = "string", implementation = String::class)
```

Which _should_ change the implementation type to look like a string instead of a `BigDecimal`.

**Comment by ilopmar** [comment]
The fix is included in micronaut-openapi 3.0.3. It will be included by default in Micronaut 3.0.2, in the meantime you can force that specific version.
